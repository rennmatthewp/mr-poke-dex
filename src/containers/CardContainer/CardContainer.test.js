import React, { Component } from 'react';
import { CardContainer, mapDispatchToProps } from './index';
import { shallow } from 'enzyme';

describe('CardContainer', () => {
  let renderedCardContainer;
  let mockPopulateTypes;
  beforeEach(() => {
    mockPopulateTypes = jest.fn();
    renderedCardContainer = shallow(
      <CardContainer poplulateTypes={mockPopulateTypes} />
    );
  });

  it('should match the snapshot', () => {
    expect(renderedCardContainer).toMatchSnapshot();
  });

  it('should call the dispatch function when using a fn from MDTP', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.populateTypes();
    expect(mockDispatch).toHaveBeenCalled();
  });
});
