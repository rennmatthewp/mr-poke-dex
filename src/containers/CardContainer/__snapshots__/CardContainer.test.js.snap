// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    poplulateTypes={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "alt": "pika piii!!",
      "src": "../../../loading.gif",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "pika piii!!",
        "src": "../../../loading.gif",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
